Enhance Crisis Support with GeoIP-based helpline detection

tilakjain619 opened the pull request:
- [x] Added get_user_country() to detect user location using IP address
- [x] Integrated country-wise crisis helplines (IN, US, GB, AU) in sidebar
- [x] Implemented fallback link to IASP global helpline directory
- [x] Displayed helplines dynamically based on detected country
- [x] Added requests package to requirements.txt
- [x] Linked to Issue #6 - Closes #6

tilakjain619 commented:
#### *Merge Conflicts Resolved* 
Updated the sidebar and solved the merge conflicts.

### Sidebar rendering with location-specific helplines  
<img width="1411" height="883" alt="Sidebar showing helplines for users in India" src="https://github.com/user-attachments/assets/5ebbbd1c-e501-493e-a797-b6683edfb9f5" />

tilakjain619 commented:
### Merge Conflicts Resolved
Updated the sidebar and solved the merge conflicts.

eccentriccoder01 commented:
Hi @tilakjain619
The current deployment from main shows this in the Crisis Support section:
<img width="380" height="614" alt="image" src="https://github.com/user-attachments/assets/dca886f9-683f-46d6-8ae9-ac5ddb08331f" />

Your PR's deployment shows just this:
<img width="343" height="529" alt="image" src="https://github.com/user-attachments/assets/13666690-d225-4a60-9b0b-0070c8392373" />

Could you elaborate a bit on how your pushed changes work? It's a bit unclear to me.

tilakjain619 commented:
The current IP detection is picking up the server location (Streamlit’s US-based host, that's why it's showing US contacts), not the user’s actual IP. I’ll update this with browser-side geolocation or a country override. Thanks for pointing it out!

eccentriccoder01 commented:
Alright @tilakjain619, also please stay updated with the latest changes in the upstream, as the main is constantly being updated by multiple people. Thanks!

tilakjain619 commented:
I've resolved the merge conflicts. I updated get_user_country() to use user’s actual browser location via streamlit-js-eval. Here is deployment link: https://crisis-support-lgu5kyqwwvhwna64yugbkn.streamlit.app/

Updated Emergency Contact section:
<img width="700" height="658" alt="image" src="https://github.com/user-attachments/assets/ed9385c8-95c2-46fd-b8dd-4d3a3f818176" />
